src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -101,8 +113,8 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
                     if (con.is_Address_Override__c)
                         // track which address's need their dates updated
                         mapAddrIdIsOverride.put(con.Current_Address__c, true);
-                } else if (isContactAddressEmpty(con)) {
-                	// if the contact has no address specified, pick up the hh default.
+                } else if (isContactAddressEmpty(con) && acctHasAddress) {
```

Comment:
Yes, this was pretty much the fix for the issue, the address reset creates a blank default address when one is not found and sets it on the contact which was what was causing it to blank out.